modules/dataproc_provisioner/variables.tf

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "name" {
+  description = "Name of the profile"
+  type        = string
+}
+
+variable "label" {
+  description = "Label of the profile"
+  type        = string
+}
+
+variable "project" {
+  description = "ID of the project hosting the cluster"
+  type        = string
+  default     = "auto-detect"
+}
+
+variable "region" {
+  description = "Region of the dataproc cluster"
+  type        = string
+  default     = "auto-detect"
+}
+
+variable "network" {
+  description = "Network to use for the cluster"
+  type        = string
+  default     = "default"
+}
+
+variable "network_host_project_id" {
+  description = "Project ID of the network"
+  type        = string
+  default     = ""
+}
+
+variable "subnet" {
+  description = "Subnet to use for the cluster"
+  type        = string
+  default     = ""
+}
+
+variable "service_account" {
+  description = "Service account to run the cluster as"
+  type        = string
+  default     = ""
+}
+
+variable "gcs_bucket" {
+  description = "Bucket to write cluster job and data data"
+  type        = string
+  default     = ""
+}
+
+variable "account_key" {
+  description = "Service account key to use for authentication"
+  type        = string
+  default     = ""
+}
+
+variable "image_version" {
+  description = "Dataproc image version to use for the cluster"
+  type        = string
+  default     = ""
```

Comment:
Please set default to `1.3-debian9`
Note that 1.4.x dataproc images are not compatible with CDAP (because they use spark 2.4 and cdap requires 2.3)

Reply:
So I think CDF is smart enough to fill this in with their own default that is compatible, which I am guessing will change overtime. Should we code this in still as it will likely get stale?

Reply:
good point, though Dataproc Provisioner does NOT check that you provide a CDF compatible image here, so it is tough to say how "smart" it will be.

Reply:
Right, I don't think it will be able to prevent you from setting a bad image, but it won't pick a bad one by default.